Common test configuration errors when gz-common5 components are missing

## Environment
* OS Version: macOS 12.6.7-x86_64
* Source or binary build? source build of `gz-physics6` as of 815896de2e83a91152ccd7e1c9f5b14846d9832e

## Description
* Expected behavior: if a dependency for a component is missing, it should skip that component and build what it can
* Actual behavior: if some gz-common5 components are not available, the build will fail with configuration errors related to the common tests

## Steps to reproduce
<!-- Provide steps so we can try to reproduce this issue -->

1. Remove one of the gz-common5 components required by dartsim or bullet, such as `geospatial`, `graphics`, or `profiler`.
2. Configure `gz-physics6` with `mkdir build; cd build; cmake ..`
3. Observer configuration errors

## Output

~~~
-- ====== Finding Dependencies ======
CMake Warning at /usr/local/share/cmake/gz-cmake3/cmake3/GzFindPackage.cmake:243 (find_package):
  By not providing "Findgz-common5.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "gz-common5", but CMake did not find one.

  Could not find a package configuration file provided by "gz-common5" with
  any of the following names:

    gz-common5Config.cmake
    gz-common5-config.cmake

  Add the installation prefix of "gz-common5" to CMAKE_PREFIX_PATH or set
  "gz-common5_DIR" to a directory containing one of the above files.  If
  "gz-common5" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  CMakeLists.txt:40 (gz_find_package)


-- Looking for gz-common5 - not found
...
-- -------------------------------------------

CMake Warning at /usr/local/share/cmake/gz-cmake3/cmake3/GzConfigureBuild.cmake:68 (message):
   CONFIGURATION WARNINGS:
   -- Skipping component [heightmap]: Missing dependency [gz-common5] (Components: geospatial, graphics, profiler).
      ^~~~~ Set SKIP_heightmap=true in cmake to suppress this warning.
   
   -- Skipping component [mesh]: Missing dependency [gz-common5] (Components: geospatial, graphics, profiler).
      ^~~~~ Set SKIP_mesh=true in cmake to suppress this warning.
   
   -- Skipping component [dartsim]: Missing dependency [gz-common5] (Components: geospatial, graphics, profiler).
      ^~~~~ Set SKIP_dartsim=true in cmake to suppress this warning.
   
   -- Skipping component [tpe]: Missing dependency [gz-common5] (Components: geospatial, graphics, profiler).
      ^~~~~ Set SKIP_tpe=true in cmake to suppress this warning.
   
   -- Skipping component [tpelib]: Missing dependency [gz-common5] (Components: geospatial, graphics, profiler).
      ^~~~~ Set SKIP_tpelib=true in cmake to suppress this warning.
   
   -- Skipping component [bullet]: Missing dependency [gz-common5] (Components: geospatial, graphics, profiler).
      ^~~~~ Set SKIP_bullet=true in cmake to suppress this warning.
   
Call Stack (most recent call first):
  CMakeLists.txt:103 (gz_configure_build)


-- Configuring library: gz-physics7
-- Adding 7 UNIT tests
-- Adding 14 INTEGRATION tests
-- Adding 2 PERFORMANCE tests
-- No tests have been specified for REGRESSION
-- Configuring library: gz-physics7-sdf
-- Skipping the component [heightmap] because the following packages are missing: gz-common5
-- Skipping the component [mesh] because the following packages are missing: gz-common5
-- Skipping the component [dartsim] because the following packages are missing: gz-common5
-- Skipping the component [tpe] because the following packages are missing: gz-common5
-- Skipping the component [bullet] because the following packages are missing: gz-common5
-- Configuring library: gz-physics7-bullet-featherstone
-- Configuring library: gz-physics7-bullet-featherstone-plugin
-- Adding codecheck target
-- Build configuration successful
-- Build type: RelWithDebInfo
-- Install prefix: /usr/local
-- Configuring done (1.1s)
CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-bullet-plugin>

  No target "gz-physics7-bullet-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:68 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-dartsim-plugin>

  No target "gz-physics7-dartsim-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:72 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-tpe-plugin>

  No target "gz-physics7-tpe-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:74 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-bullet-plugin>

  No target "gz-physics7-bullet-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:68 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-dartsim-plugin>

  No target "gz-physics7-dartsim-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:72 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-tpe-plugin>

  No target "gz-physics7-tpe-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:74 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-bullet-plugin>

  No target "gz-physics7-bullet-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:68 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-dartsim-plugin>

  No target "gz-physics7-dartsim-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:72 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-tpe-plugin>

  No target "gz-physics7-tpe-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:74 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-bullet-plugin>

  No target "gz-physics7-bullet-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:68 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-dartsim-plugin>

  No target "gz-physics7-dartsim-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:72 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-tpe-plugin>

  No target "gz-physics7-tpe-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:74 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-bullet-plugin>

  No target "gz-physics7-bullet-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:68 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-dartsim-plugin>

  No target "gz-physics7-dartsim-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:72 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-tpe-plugin>

  No target "gz-physics7-tpe-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:74 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-bullet-plugin>

  No target "gz-physics7-bullet-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:68 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-dartsim-plugin>

  No target "gz-physics7-dartsim-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:72 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-tpe-plugin>

  No target "gz-physics7-tpe-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:74 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-bullet-plugin>

  No target "gz-physics7-bullet-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:68 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-dartsim-plugin>

  No target "gz-physics7-dartsim-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:72 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-tpe-plugin>

  No target "gz-physics7-tpe-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:74 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-bullet-plugin>

  No target "gz-physics7-bullet-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:68 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-dartsim-plugin>

  No target "gz-physics7-dartsim-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:72 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-tpe-plugin>

  No target "gz-physics7-tpe-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:74 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-bullet-plugin>

  No target "gz-physics7-bullet-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:68 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-dartsim-plugin>

  No target "gz-physics7-dartsim-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:72 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-tpe-plugin>

  No target "gz-physics7-tpe-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:74 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-bullet-plugin>

  No target "gz-physics7-bullet-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:68 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-dartsim-plugin>

  No target "gz-physics7-dartsim-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:72 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-tpe-plugin>

  No target "gz-physics7-tpe-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:74 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-bullet-plugin>

  No target "gz-physics7-bullet-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:68 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-dartsim-plugin>

  No target "gz-physics7-dartsim-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:72 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-tpe-plugin>

  No target "gz-physics7-tpe-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:74 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-bullet-plugin>

  No target "gz-physics7-bullet-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:68 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-dartsim-plugin>

  No target "gz-physics7-dartsim-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:72 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:26 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:gz-physics7-tpe-plugin>

  No target "gz-physics7-tpe-plugin"
Call Stack (most recent call first):
  test/common_test/CMakeLists.txt:74 (configure_common_test)


CMake Error at test/common_test/CMakeLists.txt:49 (target_link_libraries):
  Target "COMMON_TEST_added_mass" links to:

    gz-common::gz-common

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at test/common_test/CMakeLists.txt:49 (target_link_libraries):
  Target "COMMON_TEST_addexternalforcetorque" links to:

    gz-common::gz-common

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at test/common_test/CMakeLists.txt:49 (target_link_libraries):
  Target "COMMON_TEST_basic_test" links to:

    gz-common::gz-common

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at test/common_test/CMakeLists.txt:49 (target_link_libraries):
  Target "COMMON_TEST_collisions" links to:

    gz-common::gz-common

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at test/common_test/CMakeLists.txt:49 (target_link_libraries):
  Target "COMMON_TEST_construct_empty_world" links to:

    gz-common::gz-common

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at test/common_test/CMakeLists.txt:49 (target_link_libraries):
  Target "COMMON_TEST_free_joint_features" links to:

    gz-common::gz-common

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at test/common_test/CMakeLists.txt:49 (target_link_libraries):
  Target "COMMON_TEST_joint_features" links to:

    gz-common::gz-common

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at test/common_test/CMakeLists.txt:49 (target_link_libraries):
  Target "COMMON_TEST_kinematic_features" links to:

    gz-common::gz-common

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at test/common_test/CMakeLists.txt:49 (target_link_libraries):
  Target "COMMON_TEST_link_features" links to:

    gz-common::gz-common

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at test/common_test/CMakeLists.txt:49 (target_link_libraries):
  Target "COMMON_TEST_shape_features" links to:

    gz-common::gz-common

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at test/common_test/CMakeLists.txt:49 (target_link_libraries):
  Target "COMMON_TEST_simulation_features" links to:

    gz-common::gz-common

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at test/common_test/CMakeLists.txt:49 (target_link_libraries):
  Target "COMMON_TEST_world_features" links to:

    gz-common::gz-common

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at test/plugins/CMakeLists.txt:3 (target_link_libraries):
  Target "MockDoublePendulum" links to:

    gz-common::core

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at bullet-featherstone/CMakeLists.txt:19 (target_link_libraries):
  Target "gz-physics7-bullet-featherstone-plugin" links to:

    gz-common::gz-common

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



-- Generating done (0.4s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
~~~

--- a/test/common_test/CMakeLists.txt
+++ b/test/common_test/CMakeLists.txt
@@ -24,21 +24,23 @@ set(TEST_INSTALL_DIR ${CMAKE_INSTALL_LIBEXECDIR}/gz/${GZ_DESIGNATION}${PROJECT_V
 include(GzPython)
 
 function(configure_common_test PHYSICS_ENGINE_NAME test_name)
-  set(target_name ${test_name}_${PHYSICS_ENGINE_NAME})
-  add_test(NAME ${target_name}
-    COMMAND
-      ${test_name}
-      $<TARGET_FILE:${PROJECT_LIBRARY_TARGET_NAME}-${PHYSICS_ENGINE_NAME}-plugin>
-      --gtest_output=xml:${CMAKE_BINARY_DIR}/test_results/${target_name}.xml
-  )
+  if(NOT SKIP_${PHYSICS_ENGINE_NAME} AND NOT INTERNAL_SKIP_${PHYSICS_ENGINE_NAME})
+    set(target_name ${test_name}_${PHYSICS_ENGINE_NAME})
+    add_test(NAME ${target_name}
+      COMMAND
+        ${test_name}
+        $<TARGET_FILE:${PROJECT_LIBRARY_TARGET_NAME}-${PHYSICS_ENGINE_NAME}-plugin>
+        --gtest_output=xml:${CMAKE_BINARY_DIR}/test_results/${target_name}.xml
+    )
 
-  set_tests_properties(${target_name} PROPERTIES TIMEOUT 240)
+    set_tests_properties(${target_name} PROPERTIES TIMEOUT 240)
 
-  if(Python3_Interpreter_FOUND)
-    # Check that the test produced a result and create a failure if it didn't.
-    # Guards against crashed and timed out tests.
-    add_test(check_${target_name} ${Python3_EXECUTABLE} ${GZ_CMAKE_TOOLS_DIR}/check_test_ran.py
-      ${CMAKE_BINARY_DIR}/test_results/${target_name}.xml)
+    if(Python3_Interpreter_FOUND)
+      # Check that the test produced a result and create a failure if it didn't.
+      # Guards against crashed and timed out tests.
+      add_test(check_${target_name} ${Python3_EXECUTABLE} ${GZ_CMAKE_TOOLS_DIR}/check_test_ran.py
+        ${CMAKE_BINARY_DIR}/test_results/${target_name}.xml)
+    endif()
   endif()
 endfunction()
 
@@ -66,12 +68,8 @@ foreach(test ${tests})
 
   install(TARGETS ${test_executable} DESTINATION ${TEST_INSTALL_DIR})
 
-  if (${BULLET_FOUND})
-    configure_common_test("bullet" ${test_executable})
-    configure_common_test("bullet-featherstone" ${test_executable})
-  endif()
-  if (${DART_FOUND})
-    configure_common_test("dartsim" ${test_executable})
-  endif()
+  configure_common_test("bullet" ${test_executable})
+  configure_common_test("bullet-featherstone" ${test_executable})
+  configure_common_test("dartsim" ${test_executable})
   configure_common_test("tpe" ${test_executable})
 endforeach()

